nxdrive/engine/dao/sqlite.py

```diff
@@ -2270,6 +2313,78 @@ def get_dt_uploads_raw(self, limit: int = 1) -> List[Dict[str, Any]]:
             ).fetchall()
         ]
 
+    def get_sessions(self) -> List[Session]:
+        """
+        Retrieve all Direct Transfer sessions.
+        Return a Session object.
+        """
+        con = self._get_read_connection()
+        c = con.cursor()
+        return [
+            Session(
+                res.uid,
+                res.remote_path,
+                res.remote_ref,
+                TransferStatus(res.status),
+                res.uploaded,
+                res.total,
+            )
+            for res in c.execute("SELECT * FROM Sessions")
+        ]
+
+    def get_session(self, uid: int) -> Optional[Session]:
+        """
+        Get a session by its uid.
+        """
+        con = self._get_read_connection()
+        c = con.cursor()
+        res = c.execute("SELECT * FROM Sessions WHERE uid = ?", (str(uid))).fetchone()
```

Comment:
Let SQLite manage the cast, and you have to pass a tuple of arguments:
```suggestion
        res = c.execute("SELECT * FROM Sessions WHERE uid = ?", (uid,)).fetchone()
```